content/java/concepts/hash-map/hash-map.md

```diff
@@ -0,0 +1,217 @@
+
+```java
+// Create a HashMap which stores courses and their teachers as a key/value pair
+HashMap<String, String> courseTeacher = new HashMap<String, String>();
+```
+
+## Creating a HashMap
+
+In Java, a `HashMap` can be created in the following way:
+
+````java
+HashMap<KeyDatatype, ValueDatatype> HashMapName = new HashMap<KeyDatatype, ValueDatatype>();
+
+```java
+// Import the HashMap class
+import java.util.HashMap;
+
+  // Create a HashMap which stores courses and their teachers as a key/value pair
+HashMap<String, String> courseTeacher = new HashMap<String, String>();
+````
+
+Here, `courseTeacher` will store keys of courses with `String` type and values of Teacher assigned for the course also of `String` type.
+
+```java
+// Import the HashMap class
+import java.util.HashMap;
+
+// Create a HashMap which stores students and their marks as a key-value pair
+HashMap<String, Integer> studentMarks = new HashMap<String, Integer>();
+```
+
+Here, `studentMarks` stores keys of student names of `String` type and values of marks secured by the student of `Integer` type.
+
+## Key-Value
+
+A key can be used to uniquely identify a value in a `HashMap`. This allows for efficient data storage and easy access. In the example below, the course names are the keys and the teachers assigned are the values that can be accessed using the corresponding key.
+
+```java
+// Import the HashMap class
+import java.util.HashMap;
+
+public class Main {
+  public static void main(String[] args) {
+
+    HashMap<String, String> courseTeacher = new HashMap<String, String>();
+
+    // Add keys and values (CourseNames, Teacher)
+    courseTeacher.put("History", "Ben");
+    courseTeacher.put("Mathematics", "Jeanette");
+    courseTeacher.put("Physics", "Lily");
+
+    System.out.println(courseTeacher.get("Physics"););
+    // Output: Lily
+
+    System.out.println(courseTeacher.get("History"););
+    // Output: Ben
+  }
+}
+```
+
+## Adding Items
+
+In Java, an item can be added to a `HashMap` using the `.put()` method.
+
+The `.put()` method takes in 2 attributes:
+
+- key
+- value
+
+```java
+// Import the HashMap class
+import java.util.HashMap;
+
+public class Main {
+  public static void main(String[] args) {
+    HashMap<String, String> courseTeacher = new HashMap<String, String>();
+
+    // Add keys and values (CourseNames, Teacher)
+    courseTeacher.put("History", "Ben");
+    courseTeacher.put("Mathematics", "Jeanette");
+    courseTeacher.put("Physics", "Lily");
+
+    System.out.println(courseTeacher);
+    //Output: {Mathematics=Jeanette, History=Ben, Physics=Lily}
+  }
+}
+```
+
+## Accessing an Item
+
+In Java, an item can be accessed within a `HashMap` using the `.get()` method.
+
+The `.get()` method takes in one attribute which is the key and returns the corresponding value.
+
+```java
+courseTeacher.get("Physics");
+// Returns "Lily" which is the value for the key "Physics"
```

Comment:
Can we move some of the code from lines 52-73 in here such as the `import` statement and the `main` method, and the `courseTeacher` declaration?

```java
import java.util.HashMap;

public class Main {
  public static void main(String[] args) {
    HashMap<String, String> courseTeacher = new HashMap<String, String>();

    courseTeacher.put("Physics", "Lily");
    courseTeacher.get("Physics"); // Returns "Lily"
  }
}
```